### PR TITLE
AddBillingSetup rework

### PR DESCRIPTION
**Pull request by AndrewMBurke**

### CODE REVIEW

File: examples/Billing/AddBillingSetup.cs [hunk 1]
```diff
@@ -27,6 +27,8 @@ namespace Google.Ads.GoogleAds.Examples.V4
     /// This example creates a billing setup for a customer. A billing setup is a link between
     /// payment account and customer. The new billing setup can either reuse an existing payments
```
Comment (PierrickVoulet) [inline examples/Billing/AddBillingSetup.cs:28]: "payment**s** account"

File: examples/Billing/AddBillingSetup.cs [hunk 2]
```diff
@@ -61,18 +63,15 @@ public static void Main(string[] args)
...
-                       "a link between payment account and customer. The new billing setup can " +
-                       "either reuse an existing payments account, or create a new payments " +
-                       "account with a given payments profile.\n" +
-                       "In the case of consolidated billing, a payments account is linked to the " +
-                       "manager account and is linked to a customer account via a billing setup.";
-            }
-        }
+        public override string Description =>
+            "This example creates a billing setup for a customer. A billing setup is " +
+            "a link between payment account and customer. The new billing setup can " +
```
Comment (PierrickVoulet) [inline examples/Billing/AddBillingSetup.cs:68]: "payment**s** account", "and **a** customer"

File: examples/Billing/AddBillingSetup.cs [hunk 3]
```diff
@@ -172,38 +178,40 @@ public override string Description
...
             {
-                // If there are any existing approved billing setups, the new billing setup can
-                // start immediately after the last.
+                // Retrieves the ending date time of the last billing setup.
                 string lastEndingDateTimeString = searchResponse.First().BillingSetup.EndDateTime;
 
-                // Check if the existing billing setup has no end date (i.e., is set to run
-                // indefinitely).
+                // If the ending date is null then it is set to run indefinitely. Creating a new
+                // billing setup can impact it.
```
Comment (PierrickVoulet) [inline examples/Billing/AddBillingSetup.cs:214]: I believe "Creating a new billing setup can impact it" can be misleading. I suggest we remove it and add something like "Billing setups cannot overlap each other" in the conditional block below to better explain the issue.

File: examples/Billing/AddBillingSetup.cs [hunk 4]
```diff
@@ -172,38 +178,40 @@ public override string Description
...
                 if (lastEndingDateTimeString == null)
                 {
                     throw new Exception("Cannot set ending date time for the new " +
```
Comment (PierrickVoulet) [inline examples/Billing/AddBillingSetup.cs:217]: "Cannot set **starting** date time"

File: examples/Billing/AddBillingSetup.cs [hunk 5]
```diff
@@ -212,13 +220,16 @@ FROM billing_setup
...
+
+                // Sets the new billing setup to end one day after the starting date time.
                 billingSetup.EndDateTime = lastEndingDateTime.AddDays(2).ToString("yyyy-MM-dd");
             }
             else
             {
-                // If there are no existing approved billing setups, the only acceptable start time
-                // is TimeType.Now.
+                // Otherwise, the only acceptable start time is TimeType.Now.
                 billingSetup.StartTimeType = TimeType.Now;
```
Comment (PierrickVoulet) [inline examples/Billing/AddBillingSetup.cs:233]: Make sure to set billingSetup.EndDateTime in this else block too with `TimeType.Now + 1 day`

File: examples/Billing/AddBillingSetup.cs (hunk 1 above)
Comment (AndrewMBurke) [reply examples/Billing/AddBillingSetup.cs:28]: Fixed.

File: examples/Billing/AddBillingSetup.cs (hunk 2 above)
Comment (AndrewMBurke) [reply examples/Billing/AddBillingSetup.cs:68]: Fixed.

File: examples/Billing/AddBillingSetup.cs (hunk 5 above)
Comment (AndrewMBurke) [reply examples/Billing/AddBillingSetup.cs:233]: Fixed.

File: examples/Billing/AddBillingSetup.cs (hunk 3 above)
Comment (AndrewMBurke) [reply examples/Billing/AddBillingSetup.cs]: Agreed, it felt ambiguous.  Reworded.

File: examples/Billing/AddBillingSetup.cs (hunk 4 above)
Comment (AndrewMBurke) [reply examples/Billing/AddBillingSetup.cs]: Fixed.

**Pull request merged**